### PR TITLE
Depend on Mixlib-install for install logic

### PR DESCRIPTION
**Pull request by wjordan**
This branch addresses #46 by depending on the new [mixlib-install](https://github.com/chef/mixlib-install) library to provide the install.sh implementation, as suggested by @lamont-granquist in his [comment](https://github.com/hw-cookbooks/omnibus_updater/issues/46#issuecomment-121718864). The changes are substantial so I'm just sharing this PR now for feedback/discussion and in hopes others might find this immediately useful.
- The latest alpha version of `mixlib-install` is installed via `chef_gem`.
- To replace the platform-detection duplication in `OmnibusTrucker#collect_attributes`, the `platform_detection.sh` portion of the install script is invoked in a subprocess (via backticks ``).
  - Should this make use of [mixlib-shellout](https://github.com/chef/mixlib-shellout) instead for best compatibility?
  - I assume the Ohai attributes can't be used directly with the Omnitruck API here for legacy purposes?
- Added a simple LWRP that wraps the default installation recipe with a version attribute (ref. chef/mixlib-install#1). Simple basic usage examples:

```
omnibus_updater :latest
omnibus_updater "11.18.12"
omnibus_updater "12"
```
- Removed the package-downloader code path, so this version only currently supports installation via the `#install_command` provided by mixlib-install. (The upside is that this drastically simplifies the implementation!) I figure an improvement to this PR could leverage an existing package resource similar to what @lamont-granquist described in his [comment](https://github.com/hw-cookbooks/omnibus_updater/issues/46#issuecomment-121771063).
- Bumped the cookbook version to `2.0.0` because I figured that these changes would be significant enough to break expected behavior.
- `:prerelease` is not yet re-implemented, though it could be with a bit more work.
- The `exec` patch from #102 is also included in this PR since I found it helpful for my use-case, though it's not necessary and could be extracted from this PR if needed.

### CODE REVIEW

**Comment by wjordan** [comment]
Note that the changes in this PR largely reproduce the [chef-ingredient](https://github.com/chef-cookbooks/chef-ingredient) cookbook, which I assume will eventually make this community cookbook obsolete. However, `chef-ingredient` requires chef >= 12.5.0 which won't be of any help to those trying to upgrade chef-clients below that version.

**Comment by lamont-granquist** [comment]
I don't have time to review but that's the direction that the updating-of-chef-client needs to go.

If it doesn't work to rewrite this cookbook that much, then i'd be good to have an omnibus_updater2 fork.

I agree it overlaps with chef_ingredient, but I think its a good thing to do anyway.

File: recipes/installer.rb [hunk 1]
```diff
@@ -18,53 +18,29 @@
...
 
 include_recipe 'omnibus_updater'
-remote_path = node[:omnibus_updater][:full_url].to_s
+include_recipe 'omnibus_updater::chef_killer'
 
-file '/tmp/nocheck' do
-  content 'conflict=nocheck\naction=nocheck'
-  only_if { node['os'] =~ /^solaris/ }
+chef_gem "mixlib-install" do
+  version '0.8.0.alpha.6'
```
Comment (tas50) [inline recipes/installer.rb]: This should get updated for the 1.X releases

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by tas50** [comment]
For anyone interested in this work I've taken the heart of the code and moved it into a pure resource in chef-client in this PR: https://github.com/chef-cookbooks/chef-client/pull/452

This would allow anyone to use the resource in their cookbook or apply the simple attribute driven recipe. It includes the new exit code logic and the version logic is a lot simpler due to improvements to mixlib-install over the last year.

**Comment by tas50** [comment]
Closing this one out since I've reworked it into a PR to chef-client. I super appreciate everything you did here. This was a great push to making this cookbook do what everyone expects it to do.

**Pull request closed without merging**